src/form/en_ivr.jl

```diff
@@ -0,0 +1,1763 @@
+
+```
+vr_to == vr_fr - r*csr_fr + x*csi_fr
+vi_to == vi_fr - r*csi_fr - x*csr_fr
+```
+"""
+function constraint_mc_bus_voltage_drop(pm::AbstractExplicitNeutralIVRModel, nw::Int, i::Int, f_bus::Int, t_bus::Int, f_idx::Tuple{Int,Int,Int}, f_connections::Vector{Int}, t_connections::Vector{Int}, r::Matrix{<:Real}, x::Matrix{<:Real})
+    vr_fr = [var(pm, nw, :vr, f_bus)[c] for c in f_connections]
+    vi_fr = [var(pm, nw, :vi, f_bus)[c] for c in f_connections]
+
+    vr_to = [var(pm, nw, :vr, t_bus)[c] for c in t_connections]
+    vi_to = [var(pm, nw, :vi, t_bus)[c] for c in t_connections]
+
+    csr_fr = var(pm, nw, :csr, f_idx[1])
+    csi_fr = var(pm, nw, :csi, f_idx[1])
+
+    JuMP.@constraint(pm.model, vr_to .== vr_fr - r*csr_fr + x*csi_fr)
+    JuMP.@constraint(pm.model, vi_to .== vi_fr - r*csi_fr - x*csr_fr)
+end
+
+
+"""
+	function constraint_mc_branch_current_limit(
+		pm::AbstractExplicitNeutralIVRModel,
+		nw::Int,
+		f_idx::Tuple{Int,Int,Int},
+		t_idx::Tuple{Int,Int,Int},
+		f_connections::Vector,
+		t_connections::Vector,
+		c_rating::Vector{<:Real};
+		report::Bool=true
+	)
+
+For IVR models with explicit neutrals,
+imposes a bound on the current magnitude per conductor 
+at both ends of the branch (total current, i.e. including shunt contributions).
+
+```
+cr_fr^2 + ci_fr^2 <= c_rating^2
+cr_to^2 + ci_to^2 <= c_rating^2
+```
+"""
+function constraint_mc_branch_current_limit(pm::AbstractExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, t_idx::Tuple{Int,Int,Int}, f_connections::Vector, t_connections::Vector, c_rating::Vector{<:Real}; report::Bool=true)
+    cr_fr = var(pm, nw, :cr, f_idx)
+    ci_fr = var(pm, nw, :ci, f_idx)
+    cr_to = var(pm, nw, :cr, t_idx)
+    ci_to = var(pm, nw, :ci, t_idx)
+
+    cnds_finite_rating = [c for (c,r) in enumerate(c_rating) if r<Inf]
+    JuMP.@constraint(pm.model, [c in cnds_finite_rating], cr_fr[c]^2+ci_fr[c]^2 <= c_rating[c]^2)
+    JuMP.@constraint(pm.model, [c in cnds_finite_rating], cr_to[c]^2+ci_to[c]^2 <= c_rating[c]^2)
+end
+
+
+"""
+	function constraint_mc_thermal_limit_from(
+		pm::AbstractExplicitNeutralIVRModel,
+		nw::Int,
+		f_idx::Tuple{Int,Int,Int},
+		f_connections::Vector{Int},
+		rate_a::Vector{<:Real}
+	)
+
+For IVR models with explicit neutrals,
+imposes a bound on the from-side line power magnitude.
+"""
+function constraint_mc_thermal_limit_from(pm::AbstractExplicitNeutralIVRModel, nw::Int, f_idx::Tuple{Int,Int,Int}, f_connections::Vector{Int}, rate_a::Vector{<:Real})
+    vr_fr = [var(pm, nw, :vr, f_idx[1])[t] for t in f_connections]
+    vi_fr = [var(pm, nw, :vi, f_idx[1])[t] for t in f_connections]
+    cr_fr = var(pm, nw, :cr, f_idx)
+    ci_fr = var(pm, nw, :ci, f_idx)
+
+    for idx in 1:length(rate_a)
+        if rate_a[idx]<Inf
+            psw_fr_idx = JuMP.@NLexpression(pm.model,  vr_fr[idx]*cr_fr[idx] + vi_fr[idx]*ci_fr[idx])
+            qsw_fr_idx = JuMP.@NLexpression(pm.model, -vr_fr[idx]*ci_fr[idx] + vi_fr[idx]*cr_fr[idx])
```

Comment:
It looks like you wrote a generic `thermal_limit_fr` function, so the `psw_fr` and `qsw_fr` make it a little confusing, I think change to just `pf` and `qf`

also, update the docstring to indicate that the constraint is generic to edge types